chain/neutrino.go

```diff
@@ -332,16 +332,16 @@ func (s *NeutrinoClient) Rescan(startHash *chainhash.Hash, addrs []btcutil.Addre
 
 	s.clientMtx.Lock()
 	defer s.clientMtx.Unlock()
+
 	if !s.started {
 		return fmt.Errorf("can't do a rescan when the chain client " +
 			"is not started")
 	}
+
 	if s.scanning {
 		// Restart the rescan by killing the existing rescan.
 		close(s.rescanQuit)
```

Comment:
Why do we kill the rescan if it hasn't finished? Can't we potentially miss dispatching notifications for some events?

Reply:
That's a good observation! Not sure why it was originally designed like this, but would say it makes more sense to latch on to the existing one than quit the ongoing one. 

Also possible that we will miss events by just exiting an ongoing rescan without keeping the list of watched addresses around. Will see if I can trigger this situation in tests.

Reply:
Turned out to be hard to mock the backends to test this. Ended up indeed adding to an existing rescan instead of killing the ongoing one. Passing in the `Rewind` height should ensure the rescan is started from the appropriate height.